cloudconfig/userdatacfg_test.go

```diff
@@ -1164,17 +1164,16 @@ func (s *cloudinitSuite) TestProxyWritten(c *gc.C) {
 	c.Assert(err, jc.ErrorIsNil)
 
 	cmds := cloudcfg.RunCmds()
-	first := `([ ! -e /home/ubuntu/.profile ] || grep -q '.juju-proxy' /home/ubuntu/.profile) || printf '\n# Added by juju\n[ -f "$HOME/.juju-proxy" ] && . "$HOME/.juju-proxy"\n' >> /home/ubuntu/.profile`
+	first := `[ -e /etc/profile.d/juju-proxy.sh ] || printf '\n# Added by juju\n[ -f "/etc/juju-proxy.conf" ] && . "/etc/juju-proxy.conf"\n' >> /etc/profile.d/juju-proxy.sh`
```

Comment:
I definitely think this is better than what we did before. I don't quite understand why we would create '/etc/juju-proxy.conf' and then source it in /etc/profile.d/juju-proxy.sh instead of just putting the values in /etc/profile.d/* directly.

I'm also not sure whether /etc/environment is better than /etc/profile.d

I tried to do some research, but there didn't seem to be anything definitive. (And actually, the recommendation I found was to hack /etc/pam.conf because more things inherited from there.)

The big asks are that things running inside 'systemd' scripts end up with proxy values set. Have we been able to test various possibilities and make sure our solution actually solves those?

(eg, juju deploy app; will install an application and set up a systemd script to make sure that application is running, we want to make sure that app also sees the http proxy settings.)